### PR TITLE
refactor all transformations demo

### PR DESCRIPTION
**Pull request by mateusbs17**
refactor the menu script for the all transformations demo to further
allow input text options after adjust text size introduction. See [T222752](https://phabricator.wikimedia.org/)

This code:
1) Separate Articles and Controls from the Menu Factory logic
2) Add an abstraction layer for adding controls to further allow other kinds of inputs, like adjust text size

PS: this code doesn't introduce any new transforms and doesn't intend
to change functionality

### CODE REVIEW

**Comment by mateusbs17** [comment]
@berndsi and @montehurd I did some refactoring in the all transforms demo to understand it and make it easier to add more transformation controls (somewhat similar to PCS/MCS). Let me know if you have any questions or concerns.

**Comment by berndsi** [comment]
This looks very clean. Do you know what causes the ordering of the controls?

File: demo/lib/DemoControls.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import addTableCollapseControl from './TransformationControls/addTableCollapseControl.js'
+import addTextDirectionControl from './TransformationControls/addTextDirectionControl.js'
+import addThemesControl from './TransformationControls/addThemesControl.js'
+// TODO: add addLazyImageLoadControl
+import util from './util.js'
+const { expandIframeHeight } = util
+
+let ClosureManager
+
+const transformationControls = [
```
Comment (mateusbs17) [inline demo/lib/DemoControls.js]: @berndsi I don't know why the sub-sections Platform, Text direction, and Theme doesn't follow this order, but the rest of the controls seems to follow this array order.

Comment (berndsi) [reply demo/lib/DemoControls.js]: Ok, fair enough. I was just curious. Doesn't affect your PR.

**Pull request merged**